Source/PluginProcessor.cpp

```diff
@@ -143,20 +143,16 @@ void Project10PrereqAudioProcessor::processBlock (juce::AudioBuffer<float>& buff
     
     for (auto i = totalNumInputChannels; i < totalNumOutputChannels; ++i)
         buffer.clear (i, 0, buffer.getNumSamples());
-
-    
     
-    juce::dsp::AudioBlock<float> block (buffer);
-    juce::dsp::ProcessContextReplacing<float> context (block);
-    osc.process (context);
-    gain.process(context);
+    int numSamples = buffer.getNumSamples();
     
-    /*
-     Your job is to:
-        - clear the incoming buffer
-        - fill all channels in the buffer with the sinewave generated by the osc instance.
-        - your sinewave should have a gain level of -12dbFS.
-     */
+    for (int sample = 0; sample < numSamples; ++sample)
+    {
+        for (int channel = 0; channel < totalNumOutputChannels; ++channel) {
+            auto x = osc.processSample(0.0);
+            buffer.setSample(channel, sample, x * gain.getGainLinear());

```

Comment:
you need to pass the same sample to all channels.  if you listen to the output, you'll hear ticks, pops, or glitches. 
- [ ] revise

Reply:
fixed